core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java

```diff
@@ -446,12 +450,26 @@ private int getTurn(EdgeIteratorState edge, int baseNode, int prevNode, int adjN
         // Happens a lot for trunk_links
         // For _links, comparing flags works quite good, as links usually have different speeds => different flags
         if (otherContinue != null) {
-            //We are at a fork
+            // We are at a fork
             if (!InstructionsHelper.isNameSimilar(name, prevName)
                     || InstructionsHelper.isNameSimilar(otherContinue.getName(), prevName)
-                    || !prevFlag.equals(flag)
-                    || prevFlag.equals(otherContinue.getFlags())
                     || !outgoingEdgesAreSlower) {
+
+                final RoadClass roadClass = roadClassEnc.getEnum(false, edge.getFlags());
```

Comment:
It is better to use the shorter notation: `edge.get(roadClassEnc)` for these type of requests. edge.getFlags is kind of deprecated. Similar in the getSpeed method below.

Reply:
Thanks, this method was used several times in the Instruction generation. I changed these parts.